### PR TITLE
chore(localdebug): migrate transparent prerequisite task to v3

### PR DESCRIPTION
**Pull request by XiaofuHuang**

### CODE REVIEW

File: packages/fx-core/src/core/middleware/utils/debug/debugV3MigrationUtils.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+// Licensed under the MIT license.
+
+import path from "path";
+import fs from "fs-extra";
+import { MigrationContext } from "../migrationContext";
+import { CommentArray, CommentJSONValue, CommentObject, parse } from "comment-json";
+
+export async function readJsonCommentFile(
+  context: MigrationContext,
+  filePath: string
```
Comment (kuojianlu) [inline packages/fx-core/src/core/middleware/utils/debug/debugV3MigrationUtils.ts:11]: Consider to remove context and use the full path instead.

File: packages/fx-core/src/core/middleware/utils/debug/taskMigrator.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      );
+    }
+
+    task["args"]["prerequisites"] = parse(`[
+      ${newPrerequisites.join("\n  ")}
+    ]`);
+  }
+
+  if (typeof task["label"] === "string") {
+    removeDependsOnWithLabel(task["label"], context.tasks);
```
Comment (kuojianlu) [inline packages/fx-core/src/core/middleware/utils/debug/taskMigrator.ts:38]: Is necessary to remove this task from all `dependsOn`?

File: packages/fx-core/src/core/middleware/projectMigratorV3.ts [hunk 3]
```diff
@@ -493,3 +502,38 @@ export async function userdataMigration(context: MigrationContext): Promise<void
...
+  await context.fsWriteFile(
+    Constants.tasksJsonPath,
+    commentJson.stringify(tasksJsonContent, null, 4)
+  );
+
+  // Generate app.local.yml
+  const oldProjectSettings = await loadProjectSettings(context.projectPath);
+  const appYmlGenerator = new AppLocalYmlGenerator(oldProjectSettings, debugContext.appYmlConfig);
+  const appYmlString: string = await appYmlGenerator.generateAppYml();
+  await context.fsWriteFile(path.join(SettingsFolderName, Constants.appYmlName), appYmlString);
```
Comment (kuojianlu) [inline packages/fx-core/src/core/middleware/projectMigratorV3.ts:535]: When to update `app.local.yml`?

**Review by kuojianlu — APPROVED** [review]

File: packages/fx-core/src/core/middleware/utils/debug/debugMigrationContext.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import { CommentArray, CommentJSONValue } from "comment-json";
```
Comment (swatDong) [inline packages/fx-core/src/core/middleware/utils/debug/debugMigrationContext.ts:1]: file header

**Review by swatDong — APPROVED** [review]

**Review by kuojianlu — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**